pkg/initializer/initializer.go

```diff
@@ -104,32 +99,36 @@ func (e *EtcdInitializer) restoreCorruptData() error {
 
 	e.Config.RestoreOptions.BaseSnapshot = *baseSnap
 	e.Config.RestoreOptions.DeltaSnapList = deltaSnapList
+	tempRestoreOptions := *e.Config.RestoreOptions
+	tempRestoreOptions.RestoreDataDir = fmt.Sprintf("%s.%s", tempRestoreOptions.RestoreDataDir, "part")
 
-	rs := restorer.NewRestorer(store, logger)
+	logger.Infof("Removing data directory(%s) for snapshot restoration.", tempRestoreOptions.RestoreDataDir)
+	if err := os.RemoveAll(filepath.Join(tempRestoreOptions.RestoreDataDir)); err != nil {
+		return fmt.Errorf("failed to delete member directory: %v", err)
+	}
 
-	if err := rs.Restore(*e.Config.RestoreOptions); err != nil {
+	rs := restorer.NewRestorer(store, logger)
+	if err := rs.Restore(tempRestoreOptions); err != nil {
 		err = fmt.Errorf("Failed to restore snapshot: %v", err)
 		return err
 	}
-	logger.Info("Successfully restored the etcd data directory.")
-	return err
-}
 
-func removeContents(dir string) error {
-	d, err := os.Open(dir)
-	if err != nil {
-		return err
+	if err := e.removeContents(dataDir); err != nil {
+		return fmt.Errorf("failed to remove corrupt contents with restored snapshot: %v", err)
 	}
-	defer d.Close()
-	names, err := d.Readdirnames(-1)
-	if err != nil {
-		return err
+	logger.Infoln("Successfully restored the etcd data directory.")
+	return nil
+}
+
+func (e *EtcdInitializer) removeContents(dataDir string) error {
+	logger := e.Logger
+	logger.Infof("Removing data directory(%s) for snapshot restoration.", dataDir)
+	if err := os.RemoveAll(filepath.Join(dataDir)); err != nil {
+		return fmt.Errorf("failed to delete member directory: %v", err)
 	}
-	for _, name := range names {
-		err = os.RemoveAll(filepath.Join(dir, name))
-		if err != nil {
-			return err
-		}
+
+	if err := os.Rename(filepath.Join(fmt.Sprintf("%s.%s", dataDir, "part")), filepath.Join(dataDir)); err != nil {
+		return fmt.Errorf("Failed to rename temp snap directory to member directory: %v", err)
```

Comment:
Change the error message: `fmt.Errorf("failed to rename data directory %s with err: %v",dataDir, err)`